domains/primitives/runtime/src/lib.rs

```diff
@@ -168,6 +171,9 @@ pub struct CheckExtrinsicsValidityError {
     pub transaction_validity_error: TransactionValidityError,
 }
 
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct DecodeExtrinsicError(pub String);
```

Comment:
Why a String error? Instead can be a simple `Option` type since there is not extra data we need from the Error any way

Reply:
The PR also changes block preprocessor to  dump the string(it already had the `error!` log), so figured printing where the decode actually failed would be useful